src/main/java/io/vertx/core/eventbus/impl/EventBusInternal.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.eventbus.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.eventbus.EventBus;
+
+public interface EventBusInternal extends EventBus {
```

Comment:
internally we should use `start(Promise<Void>)` and `close(Promise<Void>)`

Reply:
Why? The startup process has been futurized in this PR and it is easier to follow. If we have to make start/close consistent, I would rather futurize the shutdown process too.

Reply:
internally, I'm keen on using promises because it provides control over the context and implementations don't need to care about the context to use, as usually the most outer layer will decide which context to use.

Reply:
ok